docs/spec.md

```diff
@@ -4667,8 +4667,8 @@ Given that, `results = exec(schedule, inputs)`, where:
     update_computation(results[:][result_index], updates[:][update_index])`.
   * `updated_results` is a copy of `results` with `results[:][result_index]`
     set to `updated_values[:]`.
-  * If `result_index` is out of bounds for `shape(results[:])`, the behavior
-    is implementation-defined.
+  * If `result_index` is out of bounds for `shape(results[:])`, the update is
+    discarded.
```

Comment:
How about the following formulation which defines more precisely what it means to discard an update:

```
* `exec([update_index, ...], results) = exec([...], updated_results)` where:
  * If `result_index` is in bounds for `shape(results[:])`
    * `updated_values =
      update_computation(results[:][result_index], updates[:][update_index])`.
    * `updated_results` is a copy of `results` with `results[:][result_index]`
      set to `updated_values[:]`.
  * Otherwise
    * `updated_results = results`.
```